microservices/recordings/src/Model/RawRecordingInfo.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Model;
+
+class RawRecordingInfo
+{
+    protected string $filename;
+    protected ?string $callid = null;
+    protected ?string $hashid = null;
+
+    public function __construct(
+        protected string $file,
+    ) {
+
+        $this->filename = basename($this->file);
+
+        if (preg_match("/\w+-\w+-(.*)-\w+-mix.wav/", $this->filename, $matches)) {
+            $this->callid = urldecode($matches[1]);
+            $this->hashid = substr(md5($this->callid), 0, 8);
+        }
```

Comment:
I will validate this before creating a RawRecordingInfo instance. If this condition doesn't match, it's not a recording and doesn't deserve a RawRecordingInfo. Also $this->callid and $this->hashid shouldn't be nullable. If the fille doesn't have those values, it can not be considered a recording, so callid could be part of the constructor imo.